tests/amqp_management_ut/amqp_management_ut.c

```diff
@@ -4341,6 +4346,74 @@ TEST_FUNCTION(when_copying_the_string_fails_amqp_management_set_override_status_
     amqp_management_destroy(amqp_management);
 }
 
+TEST_FUNCTION(amqp_management_fails_if_response_to_operation_is_received_before_on_send_complete_is_called)
+{
+    // arrange
+    AMQP_MANAGEMENT_HANDLE amqp_management;
+    AMQP_VALUE result;
+    int32_t status_code = 200;
+    const char* test_status_description = "my error ...";
+    uint64_t correlation_id = 0;
+
+    amqp_management = amqp_management_create(test_session_handle, "test_node");
+    (void)amqp_management_set_override_status_description_key_name(amqp_management, "xxx");
+    (void)amqp_management_open_async(amqp_management, test_on_amqp_management_open_complete, (void*)0x4242, test_on_amqp_management_error, (void*)0x4243);
```

Comment:
```suggestion
    (void)amqp_management_open_async(amqp_management, test_on_amqp_management_open_complete, (void*)0x4242, test_on_amqp_management_error, test_source_amqp_value );
```

Reply:
So the address of test_source_amqp_value is the same that is being used in this function, but they are different things. The argument in this function is just a void* address that is used as context by the caller. So we should not tie the test_source_amqp_value with that argument, despite the memory addresses chosen being the same.